### PR TITLE
News Scrapping

### PR DESCRIPTION
**Pull request by jangoral112**

### CODE REVIEW

File: src/parser/dateparser.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+    DATE_FORMAT = "%Y-%m-%d %H:%M:%S"
+
+    @classmethod
+    def parse_datetime(cls, date: datetime):
+        return date.strftime(cls.DATE_FORMAT)
+
+    @classmethod
+    def parse_struct_time(cls, date: struct_time):
+        return strftime(cls.DATE_FORMAT, date)
```
Comment (Reszkettle) [inline src/parser/dateparser.py]: I would have used dispatch decorator:

```
from multipledispatch import dispatch 
  
@dispatch(int,int) 
def product(first,second): 
    some_logic_with_ints()
  
@dispatch(float, float)
def product(first,second): 
    some_logic_with_floats()
```

File: src/parser/descriptionparser.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+
+class DescriptionParser(object):
+
+    @classmethod
+    def parse(cls, description: str):
+        return description.split(sep="<", maxsplit=1)[0]
```
Comment (Reszkettle) [inline src/parser/descriptionparser.py]: I think that it is more convenient to extract this description.split to other method f.e left_trim_html_tags and then put a result of this inside this parse method

File: src/services/service.py [hunk 3]
```diff
@@ -11,7 +23,72 @@ class FeedService(Service):
...
+        return self._news_repository.save_all(news_list)
+
+    def scrap_news_from_all_feeds(self):
+        feed_list = self._feed_repository.get_all()
+
+        news_list = []
+
+        for feed in feed_list:
+            news_from_feed = self.scrap_news_from_feed(feed)
+            news_list = news_list + news_from_feed
```
Comment (Reszkettle) [inline src/services/service.py:44]: Could be replaced with list comprehension:
`news_list = [*self.scrap_news_from_feed(feed) for feed in feed_list]` but i don't know if we can sacrifice readability a bit

File: src/services/service.py [hunk 4]
```diff
@@ -11,7 +23,72 @@ class FeedService(Service):
...
+
+    def __news_entry_to_news_entity(self, entry, feed_id):
+        description = DescriptionParser.parse(entry["summary"]) if "summary" in entry else "No description provided."
+
+        if "published_parsed" in entry:
+            publish_date = DateParser.parse_struct_time(entry["published_parsed"])
+        else:
+            publish_date = DateParser.parse_datetime(datetime.now())
+
+        news_image_link = ""
```
Comment (Reszkettle) [inline src/services/service.py]: This is to be discussed whether we remove this field from News

File: src/parser/dateparser.py (hunk 1 above)
Comment (jangoral112) [reply src/parser/dateparser.py]: Done

File: src/parser/descriptionparser.py (hunk 2 above)
Comment (jangoral112) [reply src/parser/descriptionparser.py]: Done

File: src/services/service.py (hunk 3 above)
Comment (jangoral112) [reply src/services/service.py:44]: As discussed we do not change this fragment of code.

File: src/services/service.py (hunk 4 above)
Comment (jangoral112) [reply src/services/service.py]: Field removed.

**Pull request merged**